pkg/kudoctl/util/repo/package.go

```diff
@@ -0,0 +1,326 @@
+package repo
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/bundle"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	"regexp"
+	"sigs.k8s.io/yaml"
+	"strings"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	frameworkV0FileName = "-framework.yaml"
+	versionV0FileName   = "-frameworkversion.yaml"
+	instanceV0FileName  = "-instance.yaml"
+
+	frameworkV1FileName = "framework.yaml"
+	templateV1FileNameRegex = "templates/.*.yaml"
+	paramsV1FileName = "params.yaml"
+)
+
+const apiVersion = "kudo.k8s.io/v1alpha1"
+
+type InstallCRDs struct {
+	Framework        *v1alpha1.Framework
+	FrameworkVersion *v1alpha1.FrameworkVersion
+	Instance         *v1alpha1.Instance
+}
+
+type FrameworkPackage interface {
+	GetInstallCRDs() (*InstallCRDs, error)
+}
+
+type V0Package struct {
```

Comment:
as much as I love my interface here we can drop V0 packages immediately, up for discussion. Potentially we could drop it when we convert our official repo to the new format? But we cannot do that until just before 0.3.0 release, right? 🤔